src/connectionManager.ts

```diff
@@ -0,0 +1,84 @@
+// TODO: Reference a library instead for GUID generation?
+export type GUID = string & { isGuid: true };
+export function guid(guid: string): GUID {
+  return guid as GUID;
+}
+
+export function generateGUID(): string {
+  const timestamp = new Date().getTime();
+  const randomNum = Math.floor(Math.random() * 1000000);
+  return `${timestamp}-${randomNum}`;
+}
+
+export interface Context {
+  id: GUID;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: GUID;
+  url: URL;
+  active: boolean;
```

Comment:
I believe only one connection is active at a time - if that's the case, I think we should let the `ConnectionManager` dictate which connection is active, rather than it living on the individual `Connection` (where it's possible for multiple to be `true` at once).